Runtimes/Supplemental/cmake/modules/FindSwiftCore.cmake

```diff
@@ -0,0 +1,111 @@
+
+Imported Targets
+^^^^^^^^^^^^^^^^
+
+The following :prop_tgt:`IMPORTED` TARGETS may be defined:
+
+ ``SwiftCore``
+
+Hint Variables
+^^^^^^^^^^^^^^
+
+ ``SDKROOT``
+   Set the path to the Swift SDK Root.
+   This only affects MacOS builds.
+
+ ``swift_SDKROOT``
+   Set the path to the Swift SDK installation.
+   This only affects Linux and Windows builds.
+   Apple builds always use the library provided by the SDK.
+
+Result Variables
+^^^^^^^^^^^^^^^^
+
+The module may set the following variables if `SwiftCore_DIR` is not set.
+
+ ``SwiftCore_FOUND``
+   true if core was found
+
+ ``SwiftCore_LIBRARIES`` OR ``SwiftCore_IMPLIB``
+   the libraries to be linked
+
+#]=======================================================================]
+
+# If the SwiftCore_DIR_FLAG is specified, look there instead. The cmake-generated
+# config file is more accurate, but requires that the SDK has one available.
+if(SwiftCore_DIR)
+  if(SwiftCore_FIND_REQUIRED)
+    list(APPEND args REQUIRED)
+  endif()
+  if(SwiftCore_FIND_QUIETLY)
+    list(APPEND args QUIET)
+  endif()
+  find_package(SwiftCore NO_MODULE ${args})
+  return()
+endif()
+
+include(FindPackageHandleStandardArgs)
+
+if(APPLE)
+  # When building for Apple platforms, SwiftCore always comes from within the
+  # SDK as a tbd for a shared library in the shared cache.
+  find_path(SwiftCore_INCLUDE_DIR
+      "Swift.swiftmodule/arm64e-apple-macos.swiftinterface"
+      "Swift.swiftmodule/x86_64-apple-macos.swiftinterface"
+    HINTS
+      "${CMAKE_OSX_SYSROOT}/usr/lib/swift")
+  find_library(SwiftCore_IMPLIB
+    NAMES "libswiftCore.tbd"
+    HINTS
+      "${CMAKE_OSX_SYSROOT}/usr/lib/swift")
+  add_library(SwiftCore SHARED IMPORTED GLOBAL)
+  set_target_properties(SwiftCore PROPERTIES
+    IMPORTED_IMPLIB "${SwiftCore_IMPLIB}")
+  find_package_handle_standard_args(SwiftCore DEFAULT_MSG
+    SwiftCore_IMPLIB SwiftCore_INCLUDE_DIR)
+elseif(LINUX)
+  if(SwiftCore_STATIC)
+    set(SwiftCore_INCLUDE_DIR "${Swift_SDKROOT}/usr/lib/swift_static/linux-static/Swift.swiftmodule" CACHE STRING "")
+    find_library(SwiftCore_LIBRARY
+      NAMES "libswiftCore.a"
+      HINTS "${Swift_SDKROOT}/usr/lib/swift_static/linux-static")
+    add_library(SwiftCore STATIC IMPORTED GLOBAL)
+  else()
+    set(SwiftCore_INCLUDE_DIR "${Swift_SDKROOT}/usr/lib/swift/linux/Swift.swiftmodule" CACHE STRING "")
+    find_library(SwiftCore_LIBRARY
+      NAMES "libswiftCore.so"
+      HINTS "${Swift_SDKROOT}/usr/lib/swift/linux")
+    add_library(SwiftCore SHARED IMPORTED GLOBAL)
+  endif()
+  set_target_properties(SwiftCore PROPERTIES
+    IMPORTED_LOCATION "${SwiftCore_LIBRARY}"
+    INTERFACE_INCLUDE_DIRECTORIES "${SwiftCore_INCLUDE_DIR}")
+  find_package_handle_standard_args(SwiftCore DEFAULT_MSG
+    SwiftCore_LIBRARY SwiftCore_INCLUDE_DIR)
+elseif(WIN32)
+  set(SwiftCore_INCLUDE_DIR "${Swift_SDKROOT}/usr/lib/swift_static/linux-static/Swift.swiftmodule" CACHE STRING "")
```

Comment:
This seems like a left over from iterating on the code -- if I interpret [this comment](https://github.com/swiftlang/swift/pull/81239/files#r2070816081) correctly, the line below should be enough

```suggestion
```

Reply:
Yeah, let's use `find_path` to search for the appropriate `Swift.swiftmoule`. We should prefer to search via `Swift_SDKROOT`, but then can fall back on `$ENV{SDKROOT}`. If we're building for Windows, we shouldn't be trying to find the swiftmodule from the linux-static SDK.